File: tdd/examples/java-unit-test-demo/src/main/java/com/concordusa/unittestdemo/service/MutableInteger.java
package com.concordusa.unittestdemo.service;

public class MutableInteger {

    private Integer currentValue;

    public MutableInteger(final Integer initialValue) {
        currentValue = initialValue;
    }

    public Integer getCurrentValue() {
        return currentValue;
    }

    public void add(final Integer increment) {
        currentValue = currentValue + increment;
    }

    public void subtract(final Integer decrement){
        currentValue = currentValue - decrement;
    }
}

File: practices/tdd/examples/net-core-unit-test-demo/src/UnitTestDemo/MutableInteger.cs
namespace UnitTestDemo
{
    public class MutableInteger
    {
        private int currentValue;

        public MutableInteger()
        {
            currentValue = 0;
        }

        public MutableInteger(int initialValue) 
        {
            currentValue = initialValue;
        }

        public virtual int GetCurrentValue() 
        {
            return currentValue;
        }

        public virtual void Add(int increment) 
        {
            currentValue = currentValue + increment;
        }

        public virtual void Subtract(int decrement)
        {
            currentValue = currentValue - decrement;
        }
    }
}